pennylane/measurements/shots.py

```diff
@@ -222,6 +222,22 @@ def __all_tuple_init__(self, shots: Sequence[Tuple]):
     def __bool__(self):
         return self.total_shots is not None
 
+    def __mul__(self, scalar):
+        if not isinstance(scalar, (int, float)):
+            raise TypeError("can't multiply sequence by non-int of type 'Shots'")
```

Comment:
```suggestion
            raise TypeError("Can't multiply Shots with non-integer or float type.")
```
And the test needs to be updated as well.

Reply:
So here, i can't change the spelling here because if i dont return exactly this string statement "can't multiply sequence by non-int of type 'Shots'", its gonna throw strings dont match assertion error, again i think in one of my previous commits this particular error popped up. please verify the image below:

![image](https://github.com/PennyLaneAI/pennylane/assets/79191029/4d8045fe-66bb-462b-9187-ce5cea012e07)

Now if i go and change the input it might make the whole codebase inconsistent and might lead to many unwanted error so basically i tried to return the same error without changing anything from core, but if you know some other way i will definitely try to implement it.

Reply:
See my most recent comment. If that change is made, then you will no longer get the "regex did not match" error.

Reply:
Heyy, still same error:

![image](https://github.com/PennyLaneAI/pennylane/assets/79191029/e66b3c8f-2bc5-44e2-92d3-239e4be7a526)

Reply:
So this is achievable if someone can give me the location where error message "can't multiply sequence by non-int of type 'Shots'" is binded with error status. If we change from their then i guess we can achieve what you are aiming for!!

Reply:
Hi @AnuravModak . I think you may have only updated the test without also updating the actual `Shots.__mul__ function`. Both of those need to be updated together for the test to pass. Let me know if that works.